lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -526,6 +526,11 @@ static SmallVector<TupleTypeElt, 4> decomposeIntoTupleElements(Type type) {
     return result;
   }
 
+  if (auto parenTy = dyn_cast<ParenType>(type.getPointer())) {
+    result.push_back(parenTy->getUnderlyingType());
+    return result;
+  }
+
```

Comment:
Ah, of course. I wonder how many copies of `decomposeIntoTupleElements` and similar we have lying around.